guide/phone_order_delivery_service/phone_order_delivery_service.bal

```diff
@@ -0,0 +1,218 @@
+jms:Session jmsSession = new(jmsConnection, {
+        acknowledgementMode: "AUTO_ACKNOWLEDGE"
+    });
+
+// Service endpoint
+endpoint http:Listener deliveryEP {
+    port:9091
+};
+
+@http:ServiceConfig {basePath:"/deliveryDetails"}
+
+// phone store service, which allows users to order phones online for delivery
+service<http:Service> phone_order_delivery_service bind deliveryEP {
+    // Resource that allows users to place an order for a phone
+    @http:ResourceConfig { consumes: ["application/json"],
+        produces: ["application/json"] }
+
+    sendDelivery(endpoint caller, http:Request enrichedreq) {
+        http:Response response;
+        phoneDeliver newDeliver;
+        json reqPayload;
+
+        log:printInfo(" Order Details have received from phone_store_service");
+
+        // Try parsing the JSON payload from the request
+        match  enrichedreq.getJsonPayload() {
+            // Valid JSON payload
+            json payload => reqPayload = payload;
+            // NOT a valid JSON payload
+            any => {
+                response.statusCode = 400;
+                response.setJsonPayload({"Message":"Invalid payload - Not a valid JSON payload"});
+                _ = caller -> respond(response);
+                done;
+            }
+        }
+
+        json name = reqPayload.Name;
+        json address = reqPayload.Address;
+        json contact = reqPayload.ContactNumber;
+        json phoneName = reqPayload.PhoneName;
+
+        // If payload parsing fails, send a "Bad Request" message as the response
+        if (name == null || address == null || contact == null || phoneName == null) {
+            response.statusCode = 400;
+            response.setJsonPayload({"Message":"Bad Request - Invalid payload"});
+            _ = caller -> respond(response);
+            done;
+        }
+
+        // Order details
+        newDeliver.customerName = name.toString();
+        newDeliver.address = address.toString();
+        newDeliver.contactNumber = contact.toString();
+        newDeliver.deliveryPhoneName = phoneName.toString();
+
+        // boolean variable to track the availability of a requested phone
+        boolean isPhoneAvailable;
+        // Check whether the requested phone available
+        foreach phone in phoneInventory {
+            if (newDeliver.deliveryPhoneName.equalsIgnoreCase(phone.toString())) {
+                isPhoneAvailable = true;
+                break;
+            }
+        }
+
+        json responseMessage;
+        // If the requested phone is available, then add the order to the 'OrderQueue'
+        if (isPhoneAvailable) {
+            var phoneDeliverDetails = check <json>newDeliver;
+            // Create a JMS message
+
+            jms:Message queueMessage2 = check jmsSession.createTextMessage(phoneDeliverDetails.toString());
+
+            log:printInfo("order Delivery details  added to the delivery  Queue; CustomerName: '" + newDeliver.customerName +
```

Comment:
info logs messages are inconsistent.

Shall we change 'order Delivery details  added to the delivery  Queue' to 'Order delivery details added to the delivery queue' ? Please check other logs as well.

Reply:
done